hiplan/base/app.py

```diff
@@ -33,3 +33,14 @@ def serialize(self):
     def save_to_file(self, file_name:str = "App.json"):
         with open(file_name, "w") as f:
             f.write(json.dumps(self.serialize()))
+
+    @staticmethod
+    def deserialize(file_name:str = "App.json")
```

Comment:
Serialize ia un obiect (Task, Record etc.) si returneaza o reprezentare care poate fi scrisa intr-un fisier (a.k.a serializata), in cazul nostru un dictionar. Ca sa reprezentam acel dictionar intr-un fisier folosim JSON.

Deserialize face exact opusul: ia o reprezentare serializata a unui obiect (dict) si returneaza obiectul creat dupa atributele date (observa exemplul din issue). De aici putem trage cateva concluzii:
* Metoda trebuie sa fie statica, deoarece nu se aplica niciunui obiect in sine; obiectul este creat in urma apelarii functiei.
* Metoda trebuie doar sa primeasca reprezentare serializata (dict), nu un fisier (la fel cum nici `App.serialize` nu a primit un fisier in care sa scrie.
* Fiecare clasa va trebui sa aiba un `deserialize` implementat, deoarece fiecare deserialize va apela un altul:
  * `App` apeleaza `Board.deserialize`
  * `Board` apeleaza `Record.deserialize` 
  * etc